x-pack/plugins/reporting/server/browsers/chromium/driver/screenshot_stitcher/combine.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// No types found for this package. May want to investigate an alternative with types.
+// @ts-ignore: implicit any for JS file
+import $streamToObservable from '@samverschueren/stream-to-observable';
+import { PNG } from 'pngjs';
+import * as Rx from 'rxjs';
+import { ObservableInput } from 'rxjs';
+import { map, mergeMap, reduce, switchMap, tap, toArray } from 'rxjs/operators';
+import { Logger, Screenshot, Size } from './types';
+
+// if we're only given one screenshot, and it matches the output dimensions
+// we're going to skip the combination and just use it
+const canUseFirstScreenshot = (
+  screenshots: Screenshot[],
+  size: { width: number; height: number }
+) => {
+  if (screenshots.length !== 1) {
+    return false;
+  }
+
+  const firstScreenshot = screenshots[0];
+  return (
+    firstScreenshot.dimensions.width === size.width &&
+    firstScreenshot.dimensions.height === size.height
+  );
+};
+
+/**
+ * Combines the screenshot clips into a single screenshot of size `outputDimensions`.
+ * @param screenshots - Array of screenshots to combine
+ * @param outputSize - Final output size that the screenshots should match up with
+ * @param logger - logger for extra debug output
+ */
+export function $combine(
+  screenshots: Screenshot[],
+  outputSize: Size,
```

Comment:
Changed this to only take in Size, not Dimension, since the x and y were never used.